### PR TITLE
etcdhttp: support member/leader endpoint

### PR DESCRIPTION
**Pull request by xiang90**

### CODE REVIEW

File: etcdserver/etcdhttp/client.go [hunk 1]
```diff
@@ -159,7 +159,21 @@ func (h *membersHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	switch r.Method {
 	case "GET":
-		if trimPrefix(r.URL.Path, membersPrefix) != "" {
+		switch trimPrefix(r.URL.Path, membersPrefix) {
+		case "":
```
Comment (yichengq) [inline etcdserver/etcdhttp/client.go]: put the code to set members into this case?

File: etcdserver/server.go [hunk 2]
```diff
@@ -579,6 +581,8 @@ func (s *EtcdServer) Term() uint64 { return atomic.LoadUint64(&s.raftTerm) }
 // Index, Term, Lead, Committed, Applied, LastIndex, etc.
 func (s *EtcdServer) Lead() uint64 { return atomic.LoadUint64(&s.raftLead) }
 
+func (s *EtcdServer) Leader() types.ID { return types.ID(s.Lead()) }
```
Comment (yichengq) [inline etcdserver/server.go]: combine Lead and Leader?

Comment (xiang90) [inline etcdserver/server.go]: Lead will be an interface for raft. see the comment above.

Comment (yichengq) [inline etcdserver/server.go]: ok. understand.

File: etcdserver/etcdhttp/client.go (hunk 1 above)
Comment (xiang90) [inline etcdserver/etcdhttp/client.go]: fixed.

**Comment by yichengq** [comment]
lgtm

**Pull request merged**